### PR TITLE
FEAT: MaxwellMatrix class

### PR DESCRIPTION
**Pull request by gmalinve**

### CODE REVIEW

File: pyaedt/modules/Boundary.py [hunk 1]
```diff
@@ -1592,6 +1606,32 @@ def join_parallel(self, sources, matrix_name=None, join_name=None):
...
+                        .GetChildObject(parent_name)
+                        .GetChildObject(m)
+                        .GetChildObject(s)
+                        .GetPropValue("Source")
+                    )
+                    self.sources.append({m: {s: excitations}})
+
+
 class FieldSetup(BoundaryCommon, object):
     """Manages Far Field and Near Field Component data and execution.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:1636]:
```suggestion
    """Manages far field and near field component data and execution.
```
Capitalize only proper nouns.

File: pyaedt/modules/Boundary.py [hunk 2]
```diff
@@ -1414,16 +1414,33 @@ def __init__(self, app, name, props=None, boundarytype=None):
...
+
+    @property
+    def reduced_matrices(self):
+        """List of reduced matrix groups given a parent matrix.
+
+        Returns
+        -------
+        dict
+            Dictionary of reduced matrices where key is the name of the parent matrix
+            and values is a list of reduced matrix groups.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:1428]:
```suggestion
            Dictionary of reduced matrices where the key is the name of the parent matrix
            and the values are in a list of reduced matrix groups.
```

File: pyaedt/modules/Boundary.py [hunk 3]
```diff
@@ -1607,29 +1623,96 @@ def join_parallel(self, sources, matrix_name=None, join_name=None):
...
-        self._parent_matrix = parent_name
-        self.names = []
-        self.sources = []
+        self.parent_matrix = parent_name
+        self.name = reduced_name
+        self.__sources = None
+
+    @property
+    def sources(self):
+        """List of matrix sources.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:1634]: Generally, a property doesn't have "Parameters" and "Returns" Sections. You can delete them.

File: pyaedt/modules/Boundary.py [hunk 4]
```diff
@@ -1607,29 +1623,96 @@ def join_parallel(self, sources, matrix_name=None, join_name=None):
...
+        old_source : str
+            Original name of the source to update.
+        source_type : str
+            It can either be ``Series`` or ``Parallel``.
+        new_source : str, optional
+            New name of the source to update.
+            The default value is the old source name.
+        new_excitations : str, optional
+            List of excitations to be included in matrix reduction.
+            The default values are excitations included in the source to update.
```
Comment (PipKat) [inline pyaedt/modules/Boundary.py:1676]:
```suggestion
        old_source : str
            Original name of the source to update.
        source_type : str
            Source type, which can be ``Series`` or ``Parallel``.
        new_source : str, optional
            New name of the source to update.
            The default value is the old source name.
        new_excitations : str, optional
            List of excitations to include in the matrix reduction.
            The default values are excitations included in the source to update.
```

**Review by Samuelopez-ansys — APPROVED** [review]
Good job!

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**